### ISSUE TITLE
install_pkgs does not work

### ISSUE THREAD

**Issue by uhthomas** [bazelbuild/rules_docker#1872]
# 🐞 bug report

### Affected Rule

install_pkgs

### Is this a regression?

Yes, it works fine on e15c9ebf203b7fa708e69ff5f1cdcf427d7edf6f.

### Description

`install_pkgs` used to work fine, but now it doesn't.

```
Unable to load docker image from bazel-out/darwin-fastbuild/bin/pkgs_image.build: archive/tar: invalid tar header
```

## 🔬 Minimal Reproduction

Run `bazel run //:image` in a WORKSPACE containing.

### WORKSPACE
```py3
workspace(name = "test")

load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "io_bazel_rules_docker",
    sha256 = "59d5b42ac315e7eadffa944e86e90c2990110a1c8075f1cd145f487e999d22b3",
    strip_prefix = "rules_docker-0.17.0",
    urls = ["https://github.com/bazelbuild/rules_docker/releases/download/v0.17.0/rules_docker-v0.17.0.tar.gz"],
)

load("@io_bazel_rules_docker//repositories:repositories.bzl", container_repositories = "repositories")

container_repositories()

load("@io_bazel_rules_docker//repositories:deps.bzl", container_deps = "deps")

container_deps()

load("@io_bazel_rules_docker//container:container.bzl", "container_pull")

container_pull(
    name = "io_docker_index_library_buildpack_deps",
    digest = "sha256:84c95f56515bfaa57f8951146756e4cda20a6037dff18d4779b2619ff0549687",
    registry = "index.docker.io",
    repository = "library/buildpack-deps",
    tag = "bullseye-curl",
)
```

### BUILD.bazel
```py3
load("@io_bazel_rules_docker//container:container.bzl", "container_image")
load("@io_bazel_rules_docker//docker/package_managers:download_pkgs.bzl", "download_pkgs")
load("@io_bazel_rules_docker//docker/package_managers:install_pkgs.bzl", "install_pkgs")

download_pkgs(
    name = "installables_tar",
    image_tar = "@io_docker_index_library_buildpack_deps//image:image.tar",
    packages = ["vim"],
)

install_pkgs(
    name = "pkgs_image",
    image_tar = "@io_docker_index_library_buildpack_deps//image",
    installables_tar = ":installables_tar.tar",
    installation_cleanup_commands = "rm -rf /var/lib/apt/lists/*",
    output_image_name = "pkgs_image",
)

container_image(
    name = "image",
    base = ":pkgs_image",
)
```

## 🔥 Exception or Error

<pre><code>
$ bazel run //:image
INFO: Analyzed target //:image (24 packages loaded, 139 targets configured).
INFO: Found 1 target...
ERROR: /Users/thomas/scratch/ruledockertest/BUILD.bazel:20:16: ExtractConfig image.pkgs_image.build.config failed: (Exit 1): extract_config failed: error executing command bazel-out/host/bin/external/io_bazel_rules_docker/container/go/cmd/extract_config/extract_config_/extract_config -imageTar bazel-out/darwin-fastbuild/bin/pkgs_image.build -outputConfig ... (remaining 3 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox extract_config failed: error executing command bazel-out/host/bin/external/io_bazel_rules_docker/container/go/cmd/extract_config/extract_config_/extract_config -imageTar bazel-out/darwin-fastbuild/bin/pkgs_image.build -outputConfig ... (remaining 3 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
2021/05/24 17:38:01 Unable to load docker image from bazel-out/darwin-fastbuild/bin/pkgs_image.build: archive/tar: invalid tar header
Target //:image failed to build
Use --verbose_failures to see the command lines of failed build steps.
ERROR: /Users/thomas/scratch/ruledockertest/BUILD.bazel:20:16 ImageConfig image.0.config failed: (Exit 1): extract_config failed: error executing command bazel-out/host/bin/external/io_bazel_rules_docker/container/go/cmd/extract_config/extract_config_/extract_config -imageTar bazel-out/darwin-fastbuild/bin/pkgs_image.build -outputConfig ... (remaining 3 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox extract_config failed: error executing command bazel-out/host/bin/external/io_bazel_rules_docker/container/go/cmd/extract_config/extract_config_/extract_config -imageTar bazel-out/darwin-fastbuild/bin/pkgs_image.build -outputConfig ... (remaining 3 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
INFO: Elapsed time: 0.221s, Critical Path: 0.04s
INFO: 2 processes: 2 internal.
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully

</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
$ uname -a
Darwin <redacted> 20.4.0 Darwin Kernel Version 20.4.0: Thu Apr 22 21:46:47 PDT 2021; root:xnu-7195.101.2~1/RELEASE_X86_64 x86_64
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
Build label: 4.0.0-homebrew
Build target: bazel-out/darwin-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Sat Jan 23 02:12:39 2021 (1611367959)
Build timestamp: 1611367959
Build timestamp as int: 1611367959
  </code>
</pre>

**Rules_docker version:**

<pre>
  <code>
0.17.0
  </code>
</pre>

**Anything else relevant?**

**Comment by gravypod** [comment]
This might have something to do with my tar file name code. I'll take a
look at this after work.

**Comment by gravypod** [comment]
@uhthomas I did some debugging and it doesn't seem to be from the code that I thought it was. I made this patch here: https://github.com/bazelbuild/rules_docker/pull/1874 

I think I have an idea where this is coming from but could you try this patch and let me know if this works for now? Sorry about the inconvenience this is likely causing you.

**Comment by uhthomas** [comment]
Morning @gravypod 

Thanks a ton for taking the time to look at this.

Oddly enough, 09cefcd63a35b247db52835f8069df028cf933ad does seem to work with the example I've given. I've also tested this in our repo and it also seems ok.

Just to confirm this change was what fixed it, I also tested a96b501452c55280712911d78cef9d5a8bcbd141 (tip of master as of the time of writing) which didn't work.
